testmodule/miner.nim

```diff
@@ -0,0 +1,8 @@
+import chronos
+import pkg/ethers/providers/jsonrpc
+
+
+proc mineBlocks*(provider: JsonRpcProvider, blks: int) {.async.} =
+  for i in 1..blks:
+    discard await provider.send("evm_mine")
+    await sleepAsync(1.seconds)
```

Comment:
I think you can leave out the sleep.

Reply:
I've reduced it down to 10ms to provide enough time to allow subscription to the `newHeads` event.